pyteal/compiler/subroutines.py

```diff
@@ -108,63 +112,89 @@ def spillLocalSlotsDuringRecursion(
                 # restore the local slots after returning from the subroutine. This prevents a
                 # reentry into the current subroutine from modifying variables we are currently
                 # using.
+
+                digArgs = True
+                coverSpilledSlots = False
+                uncoverArgs = False
+                if coverAvailable:
+                    digArgs = False
+                    if len(slots) < numArgs:
+                        coverSpilledSlots = True
+                    else:
+                        uncoverArgs = True
+
                 for slot in slots:
                     # spill local slots to the stack
                     before.append(TealOp(None, Op.load, slot))
 
+                    if coverSpilledSlots:
+                        # numArgs is guaranteed to be at least 2 here (since numArgs > len(slots)
+                        # and len(slots) must be at least 1 for the code to get this far), so no
+                        # need to replace this with swap if numArgs is 1
+                        before.append(TealOp(None, Op.cover, numArgs))
+
                 for _ in range(numArgs):
                     # pull the subroutine arguments to the top of the stack, above the just spilled
-                    # local slots
-
-                    # TODO: TEAL 5+, do this instead:
-                    # before.append(TealOp(None, Op.uncover, len(slots)))
-                    # or just do cover during the previous loop where slots are loaded, whichever
-                    # is more efficient
-                    before.append(
-                        TealOp(
-                            None,
-                            Op.dig,
-                            len(slots) + subroutine.argumentCount() - 1,
+                    # local slots, if needed
+
+                    stackDistance = len(slots) + numArgs - 1
+
```

Comment:
The `stackDistance` seems to be an invariant: slot length and number of args of subroutine are not modified in the loop. Can we put this ahead the for loop over `numArgs`?

Reply:
I'd rather leave that variable there, since it's the only place it's used